next/src/field/type.ts

```diff
@@ -1,47 +1,164 @@
 import type { JsfSchemaType } from '../types'
 
-/**
- * WIP type for UI field output that allows for all `x-jsf-presentation` properties to be splatted
- * TODO/QUESTION: what are the required fields for a field? what are the things we want to deprecate, if any?
- */
-export interface Field {
-  name: string
-  label?: string
-  description?: string
-  fields?: Field[]
-  // @deprecated in favor of inputType,
+export type FieldType = 'text' | 'number' | 'select' | 'file' | 'radio' | 'group-array' | 'email' | 'date' | 'checkbox' | 'fieldset' | 'money' | 'country' | 'textarea'
+
+interface BaseField {
   type: FieldType
-  inputType: FieldType
+  name: string
+  label: string
   required: boolean
+  inputType: FieldType
   jsonType: JsfSchemaType
+  errorMessage: Record<string, string>
+  schema: any
   isVisible: boolean
-  accept?: string
-  errorMessage?: Record<string, string>
-  computedAttributes?: Record<string, unknown>
+}
+
+export interface FieldOption {
+  label: string
+  value: string
+  description?: string
```

Comment:
todo:
1. The `FieldOption` is a mirror of `oneOf`, so if the `oneOf` has, `foo` inside, then the Type includes `foo` too.
2. `value` is not necessarily a string, can be anything, number, bool, even object. It's a mirror of `oneOf.const` 🪩
3. `value` can be optional too :p Look [at this example with `pattern`](https://json-schema-form.vercel.app/?path=/story/demos-fields--select-allow-create)